src/sirerrors.c

```diff
@@ -189,6 +189,7 @@ uint32_t _sir_geterror(char message[SIR_MAXERROR]) {
         if (alloc)
             _sir_safefree(&final);
 
+        SUPPRESS_CLANG_LEAK_WARNING(final);
```

Comment:
You saw a scan-build warning for this? I don't get one here... I am a huge disliker of modifying the source to squelch warnings; I would rather change the source so it doesn't complain.

This kind of thing makes it look like we're hiding bugs.